lib/utils.ts

```diff
@@ -0,0 +1,292 @@
+    NeokingdomToken: await _loadContract<NeokingdomToken>("NeokingdomToken"),
+    PriceOracle: await _loadContract<PriceOracle>("PriceOracle"),
+    RedemptionController: await _loadContract<RedemptionController>(
+      "RedemptionController"
+    ),
+    ResolutionManager: await _loadContract<ResolutionManager>(
+      "ResolutionManager"
+    ),
+    ShareholderRegistry: await _loadContract<ShareholderRegistry>(
+      "ShareholderRegistry"
+    ),
+    TokenMock: await _loadContract<TokenMock>("TokenMock"),
+    Voting: await _loadContract<Voting>("Voting"),
+  };
+}
+
+const CHAINID_TO_NAME = {
+  666666: "localhost",
+  9000: "tevmos",
+  9001: "evmos",
+} as const;
+
+const CHAINIDS = Object.keys(CHAINID_TO_NAME).map((c) => parseInt(c));
+
+export const isChainId = (
+  chainId: number
+): chainId is keyof typeof CHAINID_TO_NAME => {
+  return CHAINIDS.includes(chainId);
+};
+
+export async function getWallet(hre: HardhatRuntimeEnvironment) {
+  const ethers = hre.ethers;
+  let { chainId, name } = await hre.ethers.provider.getNetwork();
+
+  if (chainId !== 9000 && chainId !== 9001) {
+    const [signer] = await hre.ethers.getSigners();
+    return signer;
+  }
+
+  if (name === "unknown" && isChainId(chainId)) {
+    name = CHAINID_TO_NAME[chainId];
+  }
+
+  const privateKeyName = `${name.toUpperCase()}_PRIVATE_KEY`;
+  const privateKey = process.env[privateKeyName];
+
+  if (!privateKey) {
+    console.error(
+      "Cannot load private key",
+      privateKeyName,
+      "for chainId",
+      chainId
+    );
+    process.exit(1);
+  }
+
+  // Wrap the provider so we can override fee data.
+  const provider = new ethers.providers.FallbackProvider([ethers.provider], 1);
+  provider.getFeeData = async () => {
+    const fee = await ethers.provider.getFeeData();
+
+    // https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/85#issuecomment-1028435049
+    // Here is a possible workaround for ethers.js:
+    return {
+      gasPrice: parseEther("0"),
+      lastBaseFeePerGas: parseEther("0"),
+      maxFeePerGas: fee.maxFeePerGas, //parseUnits("10", "gwei"),
+      maxPriorityFeePerGas: fee.maxPriorityFeePerGas, //parseUnits("1", "gwei"),
+    };
+  };
+
+  // Create the signer for the mnemonic, connected to the provider with hardcoded fee data
+  return new ethers.Wallet(privateKey).connect(provider);
+}
+
+/* FIXME we might need this function, leave it here for now
+export async function multicall3(
+  c: ContractContext,
+  data: Promise<PopulatedTransaction>[]
+) {
+  const cd = await Promise.all(
+    data.map(async (d) => {
+      const populatedTx = await d;
+      if (!populatedTx.data) {
+        throw new Error("Calldata empty");
+      }
+      if (!populatedTx.to) {
+        throw new Error("To empty");
+      }
+      return {
+        target: populatedTx.to,
+        allowFailure: false,
+        callData: populatedTx.data,
+      };
+    })
+  );
+  console.log(cd);
+
+  return c.Multicall3.aggregate3(cd);
+}
+*/
```

Comment:
do we still need to keep a comment here?

Reply:
removed